server/services/referral.service.js

```diff
@@ -212,3 +219,24 @@ export const calculateReferralRewards = async (referrerId) =>
       },
     },
   ]);
+
+export const userIsReferredWithoutActiveReferral = async (userId) => {
+  const referral = await Referral.find({
+    userId: ObjectId(userId),
+    'reward.status': REWARD_STATUS.PENDING,
+  });
+
+  return { isReferred: referral.length !== 0, referralId: referral[0]?._id || null };
```

Comment:
I think this should work 🤷🏽‍♂️ 

```suggestion
  return referral?.[0]?._id
```